btrfs-progs: Check metadata mirrors in find-root.

djmarcin opened the pull request.

josefbacik commented:
Since mirror_num is apparently indexed starting at 1 shouldn't

mirror_num < btrfs_num_copies(&root->fs_info->mapping_tree,
     407    
-                offset, map_length)

that be <= btrfs_num_copies?

djmarcin commented:
I was doing the check first, then incrementing mirror_num if there are more mirrors to check.  I suppose incrementing first then checking the loop condition is more consistent with for and while though, so changed.

I rebased and squashed the edit, I hope github deals with that nicely (seems like it does).

djmarcin committed:
btrfs-progs: Check metadata mirrors in find-root.

Signed-off-by: David Marcin <djmarcin@google.com>
